File: backend/spec/integration/health_spec.rb
# frozen_string_literal: true

require "swagger_helper"

RSpec.describe "Health", type: :request do
  path "/api/health" do
    get "Check API health" do
      tags "Health"
      operationId "getHealth"
      produces "application/json"
      description "Returns the API status, version, and environment. Useful for liveness probes."

      response "200", "API is healthy" do
        schema "$ref" => "#/components/schemas/HealthStatus"
        run_test!
      end
    end
  end
end

File: e2e/tests/health.spec.ts
import { expect, test } from '@playwright/test';

const API_BASE_URL = process.env.API_BASE_URL || 'http://localhost:3000';

test.describe('Health Check', () => {
  test('API /api/health returns 200', async ({ request }) => {
    const response = await request.get(`${API_BASE_URL}/api/health`);
    expect(response.status()).toBe(200);

    const body = await response.json();
    expect(body).toHaveProperty('status');
  });
});
